plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -118,8 +122,9 @@ public static Optional<String> getFilterClause(TupleDomain<ColumnHandle> tupleDo
         }
     }
 
-    private static String toPredicate(String columnName, Domain domain)
+    private static String toPredicate(PinotColumnHandle pinotColumnHandle, Domain domain, boolean forHavingClause)
```

Comment:
`forHavingClause` is determined by `PinotColumnHandle`. Passing the argument seems redundant.

Reply:
Nice catch!